editoast/src/views/path/path_item_cache.rs

```diff
@@ -42,33 +48,75 @@ impl PathItemCache {
             return Ok(PathItemCache::default());
         }
 
+        // Step 1: Retrieve operational points from database using the requested identifiers
         let (trigrams, ops_uic, ops_id) = collect_path_item_ids(path_items);
-        let uic_to_ops = retrieve_op_from_uic(conn, infra_id, &ops_uic).await?;
-        let trigram_to_ops = retrieve_op_from_trigrams(conn, infra_id, &trigrams).await?;
-        let ids_to_ops = retrieve_op_from_ids(conn, infra_id, &ops_id).await?;
+        let uic_results = retrieve_op_from_uic(conn, infra_id, &ops_uic).await?;
+        let trigram_results = retrieve_op_from_trigrams(conn, infra_id, &trigrams).await?;
+        let ids_results = retrieve_op_from_ids(conn, infra_id, &ops_id).await?;
 
-        let tracks = path_items.iter().filter_map(|item| match item {
+        // Step 2: Collect all unique OPs first, deduplicating by obj_id
+        let mut ids_to_ops: HashMap<String, OperationalPointModel> = HashMap::new();
+
+        // Collect from ids_results
+        for (_, op) in ids_results {
+            ids_to_ops.entry(op.obj_id.clone()).or_insert(op);
+        }
+
+        // Collect from trigram_results
+        for ops_vec in trigram_results.into_values() {
+            for op in ops_vec {
+                ids_to_ops.entry(op.obj_id.clone()).or_insert(op);
+            }
+        }
+
+        // Collect from uic_results
+        for ops_vec in uic_results.into_values() {
+            for op in ops_vec {
+                ids_to_ops.entry(op.obj_id.clone()).or_insert(op);
+            }
+        }
```

Comment:
Can be written in a more concise way, without mutability of `ids_to_ops` if I'm not mistaken.


```suggestion
        let ids_to_ops: HashMap<String, OperationalPointModel> = ids_results
            .into_values()
            .chain(trigram_results.into_values().flatten())
            .chain(uic_results.into_values().flatten())
            .map(|op| (op.obj_id.clone(), op))
            .collect();
```

Reply:
Done in [d26c8fa7](https://github.com/OpenRailAssociation/osrd/compare/d26c8fa7b335a3d97781ee3edb1beb73fbe52666..1a0deae628bbcad5b1e5bde5f830a899d76321d0)